nbd/src/main.cpp

```diff
@@ -95,6 +95,84 @@ static void Usage() {
         << std::endl;
 }
 
+// use for record image and device info to auto map when boot
+static void AddRecord() {
+    std::fstream fs;
+    // check if exist
+    fs.open(CURVETAB_PATH, std::fstream::in);
+    if (!fs.is_open()) {
+        std::cerr << "curve-nbd: open curvetab file failed.";
+    }
+
+    std::string lineString;
+    while (getline(fs, lineString)) {
+        if (!lineString.empty() && lineString[0] != '#') {
+            std::vector<std::string> lineArray;
+            std::stringstream ss(lineString);
+            std::string str;
+            while (getline(ss, str, '\t')) {
+                lineArray.push_back(str);
+            }
+            if (lineArray.size() != 2 && lineArray.size() != 3) {
+                std::cerr << "curve-nbd: curvetab file content error.";
+                fs.close();
+                return;
+            }
+            if (lineArray[0] == nbdConfig->imgname) {
+                fs.close();
+                return;
+            }
+        }
+    }
+    fs.close();
+    // write new record
+    fs.open(CURVETAB_PATH, std::fstream::app);
```

Comment:
Is it a problem to mount multiple volumes at the same time?

Reply:
> Is it a problem to mount multiple volumes at the same time?

add flock